### DISCUSSION TITLE
Conda Python muddle

### DISCUSSION THREAD

**Discussion by hsheldon** [idaholab/moose#19993] [Q&A Getting Started] [answered]
This isn't a MOOSE problem, rather a Python problem which I believe stems from the MOOSE installation on my MacBook. For example, when trying to use matplotlib.path.Path() I get the error "Matplotlib has no attribute "path"", which makes no sense because path.py definitely exists in the location where Matplotlib was loaded from. However, I notice that I seem to have multiple versions of Matplotlib in different locations within the Miniconda3 directory, and I wonder if this is causing a problem. What's the best way to clean up this mess? Should I simply remove Miniconda3 entirely and start over?

**Comment by GiudGiud** [comment, marked as answer]
Hello

Having multiple versions of matlplotlib doesnt seem ideal. You could remove them all with conda and re-install only one, which conda/mamba will make sure is compatible with the MOOSE tools.

Otherwise, you can try importing a specific version of matplotlib in your python script, using the commands listed here:
https://stackoverflow.com/questions/6445167/force-python-to-use-an-older-version-of-module-than-what-i-have-installed-now

Guillaume

**Reply by hsheldon** [reply]
Thanks so much, removing and reinstalling with Conda worked.

**Comment by loganharbour** [comment]
Can you run
```
echo $PATH
which python
which python3
```
and paste the result?
